### PR TITLE
PPTP-1555 Put registration amendment links behind feature flag

### PR DESCRIPTION
**Pull request by kiranprakashdigital**
### Description of Work carried through

Brief description of what/why/how.

#### Check list 
 - [x] `./precheck` was executed (Integration/Component/Unit tests)
 - [x] `sbt scalafmt test:scalafmt` was executed
 - [x] User Acceptance Tests (UAT) were run locally and have passed.

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtax/registration/views/utils/ViewUtils.scala [hunk 1]
```diff
@@ -27,50 +27,60 @@ import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist.{
...
+                       call.flatMap(
+                         c =>
+                           Some(
+                             Actions(items =
+                               Seq(
+                                 ActionItem(href = c.url,
+                                            content = Text(messages("site.link.change")),
+                                            visuallyHiddenText = Some(messages(key))
+                                 )
+                               )
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/views/utils/ViewUtils.scala:55]: Won't this hide the `change` buttons on the review page as well? I think we should be making the decision as to whether to display a change link in `...views.partials.amendment.primaryContactDetails`

**Review by RyanKellyHMRC — CHANGES_REQUESTED** [review]
I think this will also affect the review page.

File: app/uk/gov/hmrc/plasticpackagingtax/registration/views/utils/ViewUtils.scala (hunk 1 above)
Comment (kiranprakashdigital) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/views/utils/ViewUtils.scala:55]: Yeah, I think you are right... Need to pass the boolean in primaryContactDetails itself. WIll fix it

Comment (kiranprakashdigital) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/views/utils/ViewUtils.scala:58]: Done

File: app/uk/gov/hmrc/plasticpackagingtax/registration/views/partials/amendment/primaryContactDetails.scala.html [hunk 2]
```diff
@@ -18,24 +18,28 @@
...
+            viewUtils.summaryListRow("primaryContactDetails.check.fullName", primaryContactDetails.name, Some(pptRoutes.AmendContactDetailsController.updateContactName()), Some(isUkCompanyPrivateBetaEnabled)),
+            viewUtils.summaryListRow("primaryContactDetails.check.jobTitle", primaryContactDetails.jobTitle, Some(pptRoutes.AmendContactDetailsController.updateJobTitle()), Some(isUkCompanyPrivateBetaEnabled)),
+            viewUtils.summaryListRow("primaryContactDetails.check.email", primaryContactDetails.email, Some(pptRoutes.AmendContactDetailsController.updateEmail()), Some(isUkCompanyPrivateBetaEnabled)),
+            viewUtils.summaryListRow("primaryContactDetails.check.phoneNumber", primaryContactDetails.phoneNumber, Some(pptRoutes.AmendContactDetailsController.updatePhoneNumber()), Some(isUkCompanyPrivateBetaEnabled)),
+            viewUtils.summaryListRow("primaryContactDetails.check.address", primaryContactDetails.address.map(viewUtils.extractAddress), Some(pptRoutes.AmendContactDetailsController.updateAddress()), Some(isUkCompanyPrivateBetaEnabled))
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/views/partials/amendment/primaryContactDetails.scala.html:42]: This feels like a strange way to go about doing this. Why not just pass `None` as that already results in the change link being suppressed?

Comment (RyanKellyHMRC) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/views/partials/amendment/primaryContactDetails.scala.html:42]: No changes should then be needed in `ViewUtils`

**Review by RyanKellyHMRC — CHANGES_REQUESTED** [review]
I think there's a more obvious way of achieving what you are trying to do.

**Review by RyanKellyHMRC — APPROVED** [review]

**Pull request merged**